Sources/PackageModel/Manifest/TargetDescription.swift

```diff
@@ -123,37 +123,31 @@ public struct TargetDescription: Equatable, Codable {
         providers: [SystemPackageProviderDescription]? = nil,
         settings: [TargetBuildSettingDescription.Setting] = [],
         checksum: String? = nil
-    ) {
+    ) throws {
         switch type {
         case .regular, .executable, .test:
-            precondition(
-                url == nil &&
-                pkgConfig == nil &&
-                providers == nil &&
-                checksum == nil
-            )
+            if url != nil { throw Error.disallowedPropertyInTarget(targetName: name, propertyName: "url") }
```

Comment:
we can potentially make this more concise with some sort of `Self.validateNontNil(value: url, error: "the error message")` etc

Reply:
I was thinking we could potentially use key paths and loop over a list of them, but for the sake of trying to get this into 5.4, I kept it simple.

Reply:
sure, its just very verbose atm and a helper func may make the code more readable. your call!